pages/cockpit/user-guide/api-designer.adoc

```diff
@@ -0,0 +1,138 @@
+
+To add an API operation:
+
+. With the API node of the model selected, click the operation in the *Operations* section of the left-hand pane. You can choose from the full list of API CRUD operations available:
+
+- *Search* creates a `/GET` operation that retrieves all instances of the API.
+- *Read* creates a `/GET` operation that retrieves a specific instance by ID.
+- *Delete* creates a `/DELETE` operation that deletes a specific instance by ID.
+- *Create* creates a `/POST` operation that creates a new instance.
+- *Update* creates a `/PUT` operation that updates a specific instance by ID.
+- *Patch* creates a `/PATCH` operation that updates elements of a specific instance by ID.
+
+. To add pagination query parameters to your operation, select the *Pagination* option.
+. In the *Optional param* section, add any optional query parameters.
++
+API Designer adds the related CRUD operation to the right-hand pane as Swagger documentation. Each new operation includes the currently defined attributes by default, as well as pagination query parameters if you enabled pagination, and any other optional query parameters you specified.
++
+image:cockpit/api-designer-operations.png[]
+
+=== Define API attributes
+
+API Designer comes with a default `id` attribute and a generic example attribute called `an attribute`. You can update or delete these default attributes, and add your own attributes.
+
+TIP: To delete an attribute, right-click on the node and select *Delete*.
+
+To add a new attribute:
+
+. Hover over the node where you want to add the attribute, until you see a plus icon on one or both sides of the node.
++
+image:cockpit/api-designer-add-att.png[]
++
+. If the node is:
+
+- the API node, click the plus icon to the left or right to create an upstream or downstream attribute.
+- an attribute node, click the plus icon to the right to create a child attribute.
+
+. In the *Attribute* section of the left-hand pane, enter the attribute label and type. You can also enter the following optional values:
+- *Description*: appears in the Swagger documentation and OAS descriptions.
+- *Example*: appears in the Swagger documentation and OAS examples.
+- *Comment*: an internal comment for your own usage.
+. In the *Options* section, select the attribute options:
+- *List*: the attribute is an array. Arrays are shown in the graphical representation between square brackets.
+- *In Abstract*: you want the attribute to be returned in 200 responses.
+- *Searchable*: you want the attribute to appear as a query parameter you can use to search for specific values.
+. In the *Optional param* section, select additional options for the attribute. For example, if you want the attribute to appear in the Swagger documentation, select *In documentation*.
++
+If you select this option, API Designer updates the Swagger documentation with details of the new attribute:
++
+image:cockpit/api-designer-attributes.png[]
+
+=== Try out your operation
+
```

Comment:
I would not mention the "Try it". As users are in the design phase, there is most probably no implementation at all of the API yet. So nothing to try. By the way, this button is meant to disappear soon